README.rst

```diff
@@ -48,3 +51,178 @@ API documentation is available here: API-docs_
 .. _hops-examples: https://github.com/logicalclocks/hops-examples
 .. _hopsML: https://hops.readthedocs.io/en/latest/hopsml/hopsML.html
 .. _API-docs: http://hops-py.logicalclocks.com/
+
+
+
+------------------------------------
+Quick Start: Python with HopsML
+------------------------------------
+
+Hops uses PySpark to distribute the execution of Python programs in a cluster. PySpark applications consist of two main components, a Driver and one to many Executors. The Driver and the Executors can be started on potentially any host in the cluster and use both the network and the HDFS filesystem to coordinate.
+
+
+Restructuring Python Programs as PySpark Programs
+--------------------------------------------------------------------
+
+If you want to run a Python program, e.g.,  to train a neural network on a GPU on Hops, you will need to restructure your code. The reason for this is that your single Python process needs to be restructured as a PySpark program, see the figure below.
+
+.. _hopsml-pyspark.png: imgs/hopsml-pyspark.png
+.. figure:: imgs/hopsml-pyspark.png
+    :alt: HopsML Python Program
+    :target: `hopsml-pyspark.png`_
+    :align: center
+    :scale: 75 %
+    :figclass: align-center
+
+The good news is that all you will need to do to get started is to move your code inside a function. In the code snippet below, the Executor code is on lines 1-3 and the Driver code is on lines 5-7. For the Executor, you define a function (e.g., *train*, but the function can have any name).  The code in the function will get run on Executors (containers). To invoke the Executor function (*train*) from the Driver (the main part of your Python program), you use the Experiment API. Launch a single Executor with *experiment.launch(<fn_name>)*.  Launch many Executors with *experiment.grid_search(<fn_name>)* for hyperparameter optimization, and *experiment.collective_all_reduce(<fn_name>)* for distributed training.
+
+
+.. code-block:: python
+  :linenos:
+   
+  def train():
+    import tensorflow as tf
+    # training code here

```

Comment:
This code block does not show in the README on Github for me when I preview it. All other code blocks are shown. Might be an indentation issue?

Reply:
@jimdowling  I tested it locally, it works with Sphinx but not for Github readme. It seems that Github do not support line-numbers in code-blocks. You can remove `:linenos:` and just hardcode the numbers as a workaround, i.e:

```
.. code-block:: python
  1. def train():
  2.  import tensorflow as tf
  3.  # training code here
  4.
  5. # Driver code starts here
  6. from hops import experiment
  7. experiment.launch(train)
```
https://github.com/Limmen/hops-util-py/tree/test